src/engine/guess_toolset.bat

```diff
@@ -32,6 +32,21 @@ if NOT "_%VS150COMNTOOLS%_" == "__" (
     set "BOOST_JAM_TOOLSET=vc141"
     set "BOOST_JAM_TOOLSET_ROOT=%VS150COMNTOOLS%..\..\VC\"
     goto :eof)
+
+set VSWHERE_PRE=Microsoft.VisualStudio.Product.
+set VSWHERE_IDS=%VSWHERE_PRE%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+set VSWHERE_REQ=-requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
+set VSWHERE_PRP=-property installationPath
+set VSWHERE="%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% set VSWHERE="%ProgramFiles%\Microsoft Visual Studio\Installer\vswhere.exe"
+if not exist %VSWHERE% goto :skip_vswhere
+for /f "usebackq tokens=*" %%i in (`%VSWHERE% -latest -products %VSWHERE_IDS% %VSWHERE_REQ% %VSWHERE_PRP%`) do (
+    set "VS150COMNTOOLS=%%i"
+    set "BOOST_JAM_TOOLSET=vc141"
+    set "BOOST_JAM_TOOLSET_ROOT=%%i\VC\"
```

Comment:
Please also add `BOOST_JAM_TOOLSET_ROOT` detection to config_toolset.bat
https://github.com/boostorg/build/blob/d2ff572c2399bc952f6a15a9fc1a01f7a82a4730/src/engine/config_toolset.bat#L168-L171

It will be used if user call `build.bat vc141`